### DISCUSSION TITLE
CUDA.dockerfile error

### DISCUSSION THREAD

**Discussion by Ryan12345678910111213** [IBM/aihwkit#510] [Installation Q&A]
When I used the CUDA.dockerfile to build images, I got errors like this:

> @LAPTOP-U0NG45HI:~/aihwkit/dockerfile$ sudo docker build -f CUDA.Dockerfile  -t aihwkit:0.1 .
> [+] Building 264.7s (6/13)
>  => [internal] load .dockerignore                                                                                  0.0s
>  => => transferring context: 2B                                                                                    0.0s
>  => [internal] load build definition from CUDA.Dockerfile                                                          0.0s
>  => => transferring dockerfile: 2.12kB                                                                             0.0s
>  => [internal] load metadata for docker.io/nvidia/cuda:11.8.0-devel-ubuntu22.04                                    0.8s
>  => [1/9] FROM docker.io/nvidia/cuda:11.8.0-devel-ubuntu22.04@sha256:cedead1946ab20934c262493ee3ea2b51d7552ff54  259.5s
>  => => resolve docker.io/nvidia/cuda:11.8.0-devel-ubuntu22.04@sha256:cedead1946ab20934c262493ee3ea2b51d7552ff54b6  0.0s
>  => => sha256:31ee91ff9a0bae4bf5880ac049a640c5a418b4a638109f0fb929e1b7e0024caa 2.63kB / 2.63kB                     0.0s
>  => => sha256:cedead1946ab20934c262493ee3ea2b51d7552ff54b65640fa145237920a7554 743B / 743B                         0.0s
>  => => sha256:7b5a6d412769f2ca6efb1493b0ac92e152de60e9acbfbb2ab3e008fbcd1d9d1e 4.62MB / 4.62MB                     1.7s
>  => => sha256:7f7a203864e09da622176128d339074ae465f42f122fae61015f1ae9e76745fa 56.23MB / 56.23MB                  23.1s
>  => => sha256:7825ced91aac831b82e966cd1c9a9049297341b7b9c4848bda9b6818c6567f1b 18.26kB / 18.26kB                   0.0s
>  => => sha256:837dd4791cdc6f670708c3a570b72169263806d7ccc2783173b9e88f94878271 29.53MB / 29.53MB                   1.9s
>  => => sha256:a02f364de9e6550724aa1a9b743690c7e29eeb8687d62e28c78332b38e702710 184B / 184B                         1.9s
>  => => extracting sha256:837dd4791cdc6f670708c3a570b72169263806d7ccc2783173b9e88f94878271                          1.0s
>  => => sha256:f6aa3af841dff731905e9b9a386cd1ffc9294e0a64f3708e197aa2463307fe24 6.89kB / 6.89kB                     2.1s
>  => => sha256:dbd59ad52ec64c5982ad6ee42a471802ea7de15fd56223f9d83b5fbd5eed22c0 1.38GB / 1.38GB                   176.5s
>  => => sha256:c597438859bb9960abd2207cc4d84cc0c472fbeaca4d4a18e91cc1d4d3ca53f0 63.70kB / 63.70kB                   2.2s
>  => => sha256:7055b728bfdbd6a9c0b2716ba0cf1a094624506d838ba1352d44f8c1e6cbb324 1.68kB / 1.68kB                     2.3s
>  => => sha256:674f79653b3aaff3d3ba1c272eb4335209ff2c7e65eb7424a9e24b2e9497d205 1.52kB / 1.52kB                     2.4s
>  => => sha256:35ec922b28680b3865ebe763b7c1ef4c6de90c52a16f2927dda35e9376b1cc8f 2.46GB / 2.46GB                   230.0s
>  => => extracting sha256:7b5a6d412769f2ca6efb1493b0ac92e152de60e9acbfbb2ab3e008fbcd1d9d1e                          0.2s
>  => => extracting sha256:7f7a203864e09da622176128d339074ae465f42f122fae61015f1ae9e76745fa                          0.8s
>  => => sha256:280e612b0e9563cd2cd740339cc35090b4d49b2ade731532a865fce0e4cad95c 88.23kB / 88.23kB                  23.7s
>  => => extracting sha256:a02f364de9e6550724aa1a9b743690c7e29eeb8687d62e28c78332b38e702710                          0.0s
>  => => extracting sha256:f6aa3af841dff731905e9b9a386cd1ffc9294e0a64f3708e197aa2463307fe24                          0.0s
>  => => extracting sha256:dbd59ad52ec64c5982ad6ee42a471802ea7de15fd56223f9d83b5fbd5eed22c0                         15.8s
>  => => extracting sha256:c597438859bb9960abd2207cc4d84cc0c472fbeaca4d4a18e91cc1d4d3ca53f0                          0.0s
>  => => extracting sha256:7055b728bfdbd6a9c0b2716ba0cf1a094624506d838ba1352d44f8c1e6cbb324                          0.0s
>  => => extracting sha256:674f79653b3aaff3d3ba1c272eb4335209ff2c7e65eb7424a9e24b2e9497d205                          0.0s
>  => => extracting sha256:35ec922b28680b3865ebe763b7c1ef4c6de90c52a16f2927dda35e9376b1cc8f                         29.0s
>  => => extracting sha256:280e612b0e9563cd2cd740339cc35090b4d49b2ade731532a865fce0e4cad95c                          0.0s
>  => [internal] load build context                                                                                  0.1s
>  => => transferring context: 2.12kB                                                                                0.0s
>  => ERROR [2/9] RUN apt-get update &&     DEBIAN_FRONTEND="noninteractive" apt-get install --yes     --no-install  4.3s
> ------
>  > [2/9] RUN apt-get update &&     DEBIAN_FRONTEND="noninteractive" apt-get install --yes     --no-install-recommends     bash     bash-completion     cmake     curl     git     libopenblas-dev     linux-headers-$(uname -r)     nano     python3 python3-dev python3-pip python-is-python3     sudo     wget &&     apt-get autoremove -y &&     apt-get clean &&     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*:
> #0 0.701 Get:1 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
> #0 0.767 Get:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1581 B]
> #0 0.828 Get:3 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
> #0 0.854 Get:4 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
> #0 0.855 Get:5 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [359 kB]
> #0 0.896 Get:6 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [108 kB]
> #0 0.933 Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages [17.5 MB]
> #0 1.352 Get:8 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [631 kB]
> #0 1.995 Get:9 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [36.3 kB]
> #0 1.998 Get:10 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [935 kB]
> #0 2.220 Get:11 http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages [1792 kB]
> #0 2.262 Get:12 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [541 kB]
> #0 2.348 Get:13 http://archive.ubuntu.com/ubuntu jammy/restricted amd64 Packages [164 kB]
> #0 2.359 Get:14 http://archive.ubuntu.com/ubuntu jammy/multiverse amd64 Packages [266 kB]
> #0 2.382 Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [42.2 kB]
> #0 2.382 Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [919 kB]
> #0 2.440 Get:17 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [545 kB]
> #0 2.469 Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1185 kB]
> #0 2.631 Get:19 http://archive.ubuntu.com/ubuntu jammy-backports/main amd64 Packages [49.4 kB]
> #0 2.730 Get:20 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [27.0 kB]
> #0 2.900 Fetched 25.6 MB in 2s (10.7 MB/s)
> #0 2.900 Reading package lists...
> #0 3.474 Reading package lists...
> #0 3.989 Building dependency tree...
> #0 4.096 Reading state information...
> #0 4.132 E: Unable to locate package linux-headers-5.15.90.1-microsoft-standard-WSL2
> #0 4.132 E: Couldn't find any package by glob 'linux-headers-5.15.90.1-microsoft-standard-WSL2'
> #0 4.132 E: Couldn't find any package by regex 'linux-headers-5.15.90.1-microsoft-standard-WSL2'
> ------
> CUDA.Dockerfile:13
> --------------------
>   12 |     # Install dependencies
>   13 | >>> RUN apt-get update && \
>   14 | >>>     DEBIAN_FRONTEND="noninteractive" apt-get install --yes \
>   15 | >>>     --no-install-recommends \
>   16 | >>>     bash \
>   17 | >>>     bash-completion \
>   18 | >>>     cmake \
>   19 | >>>     curl \
>   20 | >>>     git \
>   21 | >>>     libopenblas-dev \
>   22 | >>>     linux-headers-$(uname -r) \
>   23 | >>>     nano \
>   24 | >>>     python3 python3-dev python3-pip python-is-python3 \
>   25 | >>>     sudo \
>   26 | >>>     wget && \
>   27 | >>>     apt-get autoremove -y && \
>   28 | >>>     apt-get clean && \
>   29 | >>>     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
>   30 |
> --------------------
> ERROR: failed to solve: process "/bin/sh -c apt-get update &&     DEBIAN_FRONTEND=\"noninteractive\" apt-get install --yes     --no-install-recommends     bash     bash-completion     cmake     curl     git     libopenblas-dev     linux-headers-$(uname -r)     nano     python3 python3-dev python3-pip python-is-python3     sudo     wget &&     apt-get autoremove -y &&     apt-get clean &&     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*" did not complete successfully: exit code: 100

Can you help me to fix this error?

Other information
Pytorch version: 2.0.1 torchvision 0.15.2
Package version: aihwkit 0.7.1
OS: Ubuntu 22.04.2 LTS (GNU/Linux 5.15.90.1-microsoft-standard-WSL2 x86_64)
Python version: 3.8.17

**Comment by maljoras** [comment]
Looks like you are missing some system libraries and development packages: 
```
#0 4.132 E: Unable to locate package linux-headers-5.15.90.1-microsoft-standard-WSL2
#0 4.132 E: Couldn't find any package by glob 'linux-headers-5.15.90.1-microsoft-standard-WSL2'
#0 4.132 E: Couldn't find any package by regex 'linux-headers-5.15.90.1-microsoft-standard-WSL2'
```
